composer/core/state.py

```diff
@@ -378,6 +386,22 @@ def __init__(
         self.eval_metric_values: Dict[str, float] = {}
         self.total_loss_dict: Dict[str, float] = {}
 
+    @property
+    def train_dataloader(self):
+        """The train dataloader."""
+        return self._train_dataloader
+
+    @train_dataloader.setter
+    def train_dataloader(self, train_dataloader: Optional[Union[Iterable, DataLoader]]):
+        self._train_dataloader = train_dataloader
+        if self.dataset_state:
```

Comment:
```suggestion
        self._train_dataloader = train_dataloader
        # Load dataset state from checkpoint when train_dataloader is set
        if self.dataset_state:
```

Reply:
It would be nice to include a small comment -- it's a little hard to understand what's going on here without context of entire design